packages/core/test/id.test.ts

```diff
@@ -43,7 +43,7 @@ const idMachine = createMachine({
           }
         },
         dot: {
-          id: 'B.dot'
+          id: '.dot'
```

Comment:
that looks incorrect/confusing, I think that it should be completely forbidden to an `id` with dots in it (it doesn't matter if the dot is leading or just in the middle of it). It introduces ambiguity because it's not obvious if it should be interpreted as a literal or if it should be interpreted as a path chained from the first segment

Reply:
You're right; not sure how that snuck in here. Will review.